app/src/main/java/com/cornellappdev/android/eatery/ExpandableListAdapter.java

```diff
@@ -109,57 +111,64 @@ public View getGroupView(int i, boolean b, View view, ViewGroup viewGroup) {
     TextView timetext1 = view.findViewById(R.id.time1);
 
     try {
-      CafeteriaModel myCafe = null;
+      EateryModel eatery = null;
       int count = 0;
       while (count < cafeData.size()) {
         if (m.equals(cafeData.get(count).getNickName())) {
-          myCafe = cafeData.get(count);
+          eatery = cafeData.get(count);
         }
         count++;
       }
+      if (eatery instanceof DiningHallModel) {
+        List<MealModel> day = ((DiningHallModel) eatery).getWeeklyMenu().get(dayOfWeek);
+        int length = day.size() - 1;
+
+        //finding correct idx to get desired meal model from day array
+        MealModel meal;
+        if (length == 1) {
+          meal = day.get(mealIndex / 2);
+        } else if (length == 2) {
+          meal = day.get(mealIndex);
+        } else if (length == 3) {
+          meal = day.get(mealIndex - 2 + length);
+        } else {
+          meal = day.get(0);
+        }
 
-      ArrayList<MealModel> day = myCafe.getWeeklyMenu().get(dateOffset + 1);
-      int length = day.size() - 1;
-
-      // finding correct idx to get desired meal model from day array
-      MealModel meal;
-      if (length == 1) {
-        meal = day.get(mealIndex / 2);
-      } else if (length == 2) {
-        meal = day.get(mealIndex);
-      } else if (length == 3) {
-        meal = day.get(mealIndex - 2 + length);
-      } else {
-        meal = day.get(0);
-      }
+        DateTimeFormatter localDateFormat = DateTimeFormatter.ofPattern("h:mma");
+        ZonedDateTime now = ZonedDateTime.now();
+        Resources res = context.getResources();
+
+        String formattedEndTime = localDateFormat.format(meal.getEnd());
+        String formattedStartTime = localDateFormat.format(meal.getStart());
+
+        ZoneId cornell = TimeUtil.getInstance().getCornellTimeZone();
+        ZonedDateTime startTime = meal.getStart().atZone(cornell);
+        ZonedDateTime endTime = meal.getStart().atZone(cornell);
 
-      SimpleDateFormat localDateFormat = new SimpleDateFormat("h:mm a");
-      Date date = new Date();
-
-      String endTime = localDateFormat.format(meal.getEnd());
-      String startTime = localDateFormat.format(meal.getStart());
-
-      // meal date seems to be off by one day
-      if ((date.getTime() > meal.getStart().getTime() && date.getTime() < meal.getEnd().getTime())
-          || date.getDay() != meal.getStart().getDay()
-          || date.getTime() < meal.getStart().getTime()) {
-        timetext1.setText("Open from " + startTime + " to " + endTime);
-        timetext1.setTextColor(Color.parseColor("#1a84db"));
-      } else {
-        String mealString = "";
-        if (mealIndex == 0) {
-          mealString = "Breakfast";
-        } else if (meal.getType().equals("Brunch")) {
-          mealString = "Brunch";
-        } else if (mealIndex == 1) {
-          mealString = "Lunch";
+        //meal date seems to be off by one day
```

Comment:
same comment regarding `//`

Reply:
I'll fix it, I didn't touch any comments from previous coders.